enhancements/sig-policy/89-operator-policy-kind/README.md

```diff
@@ -0,0 +1,557 @@
+    upgradesAvailable: Condition
+    upgradesProgressing: NonCompliant
+```
+
+When `remediationAction` is set to `inform`, no actions (creates, deletes, updates) will be taken on
+any resources, but the current state may be *read* to identify if the operator is installed as
+specified. When resources are missing, the policy will be NonCompliant, with a message specifying
+what is missing, and that it will not be created because the policy is not being enforced. When it
+is set to `enforce`, OperatorGroups and Subscriptions may be created by the controller, InstallPlans
+may be approved, and resources may be removed when the policy is deleted.
+
+The optional `spec.operatorGroup` section defines an
+[OperatorGroup](https://github.com/operator-framework/api/blob/master/crds/operators.coreos.com_operatorgroups.yaml).
+If not specified, and if the operator supports this InstallMode, then it will ensure that an
+[AllNamespaces](https://olm.operatorframework.io/docs/advanced-tasks/operator-scoping-with-operatorgroups/#targetnamespaces-and-their-relationship-to-installmodes)
+OperatorGroup exists in the namespace of the subscription.
+
+The `spec.subscription` section includes the fields from the spec of an [OLM
+Subscription](https://github.com/operator-framework/api/blob/master/crds/operators.coreos.com_subscriptions.yaml).
+For more information on the `config` field, see:
+https://github.com/operator-framework/api/blob/v0.17.3/pkg/operators/v1alpha1/subscription_types.go#L42.
+The `installPlanApproval` field is optional. If the policy is in `enforce` mode and the allowed CSVs
+are restricted as specified below, then the `installPlanApproval` field on the Subscription will
+always be Manual, regardless of the setting here.
+
+The `spec.allowedCSVs` list specifies what installed versions are considered Compliant when the
+policy is in `inform` mode, and which InstallPlans can be approved when in `enforce` mode. Only
+exact matches are considered, except for a special value, "*" which matches any version. This field
+must be set, and must have at least one item.
+
+The `spec.removalBehavior` field allows configuration of what will be removed when the policy is
+deleted. Each kind that might be deleted should support `Keep` and `Delete`, and can potentially
+have additional values specific to the resources in question. For example, the OperatorGroup should
+likely only be removed if it isn't being used by another subscription. 
+
+The `spec.statusConfig` field allows the author to specify what effect specific resource statuses
+will have on the OperatorPolicy status and compliance events. The currently planned values are:
+`Condition`, which will record the status in `status.conditions` and in the compliance event; and
+`NonCompliant`, which additionally updates the `status.compliant` field to NonCompliant which can
+cause the root policy on the hub to become NonCompliant and may throw an alert. Additional values
+could be added in the future.
```

Comment:
I guess if we put `NonCompliant` for both `upgradesAvailable` and `upgradesProgressing`, it might cover 99% of the whole process. It might be hard to guarantee it will always work like that as things evolve though. Is it better to introduce an explicit field for `atLatestKnown`? This is the main use case for telco and it's part of our current du reference spec.

Reply:
I realized I didn't fully specify this, but I was expecting `upgradesAvailable` to basically correspond to the `atLatestKnown` case... is there a situation I'm missing there? I guess I don't know what the Subscription's state will look like if it's not at the newest version, but it can't upgrade to that because of a version incompatibility or something.

Reply:
Yes, there are other states in addition to these two:
`const (
	SubscriptionStateNone             = ""
	SubscriptionStateFailed           = "UpgradeFailed"
	SubscriptionStateUpgradeAvailable = "UpgradeAvailable"
	SubscriptionStateUpgradePending   = "UpgradePending"
	SubscriptionStateAtLatest         = "AtLatestKnown"
)`